pybamm/solvers/c_solvers/klu.cpp

```diff
@@ -0,0 +1,429 @@
+// pybind wrapping of sundials with the KLU linear solver
+// currently just in a quick and dirty form to get something working
+
+#include <stdio.h>
+#include <math.h>
+
+#include <ida/ida.h>                    /* prototypes for IDA fcts., consts.    */
+#include <nvector/nvector_serial.h>     /* access to serial N_Vector            */
+#include <sunmatrix/sunmatrix_sparse.h> /* access to sparse SUNMatrix           */
+#include <sunlinsol/sunlinsol_klu.h>    /* access to KLU linear solver          */
+#include <sundials/sundials_types.h>    /* defs. of realtype, sunindextype      */
+#include <sundials/sundials_math.h>     /* defs. of SUNRabs, SUNRexp, etc.      */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include <pybind11/functional.h>
+namespace py = pybind11;
+
+using residual_type = std::function<py::array_t<double>(double, py::array_t<double>, py::array_t<double>)>;
+using jacobian_type = std::function<py::array_t<double>(double, py::array_t<double>, double)>;
+using event_type = std::function<py::array_t<double>(double, py::array_t<double>)>;
+using np_array = py::array_t<double>;
+
+using jac_get_type = std::function<np_array()>;
+
+#pragma GCC visibility push(hidden)
```

Comment:
does this work on compilers other than gcc?

Reply:
This was the suggested thing to do in response to some warning I was getting when compiling. No longer getting it (not sure why) so will just remove this line.